src/test/java/de/tum/in/www1/artemis/TextExerciseIntegrationTest.java

```diff
@@ -740,6 +740,22 @@ void testInstructorGetResultsFromOwningCoursesNotEmpty() throws Exception {
         assertThat(resultNon.getResultsOnPage()).isNullOrEmpty();
     }
 
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    void testCourseAndExamFiltersAsInstructor() throws Exception {
+        database.addCourseWithOneReleasedTextExercise("Ankh");
+        database.addCourseExamExerciseGroupWithOneTextExercise("Ankh-Morpork");
+        exerciseIntegrationTestUtils.testCourseAndExamFilters("/api/text-exercises");
+    }
+
+    @Test
+    @WithMockUser(username = "admin", roles = "ADMIN")
+    void testCourseAndExamFiltersAsAdmin() throws Exception {
+        database.addCourseWithOneReleasedTextExercise("Ankh");
+        database.addCourseExamExerciseGroupWithOneTextExercise("Ankh-Morpork");
+        exerciseIntegrationTestUtils.testCourseAndExamFilters("/api/text-exercises");
+    }
```

Comment:
Please remove code duplication. My comment on https://github.com/ls1intum/Artemis/pull/5484 is valid here as well.